### PR TITLE
Make Lifetime container dispose logic more robust

### PR DESCRIPTION
**Pull request by hanabi1224**
As discussed in https://github.com/unitycontainer/unity/issues/196

### CODE REVIEW

File: src/Container/Lifetime/LifetimeContainer.cs [hunk 1]
```diff
@@ -100,12 +100,29 @@ protected virtual void Dispose(bool disposing)
...
+                        {
+                            exceptions.Add(e);
+                        }
                     }
 
                     _items.Clear();
+
+                    if (exceptions.Count == 1)
+                    {
+                        throw exceptions.First();
```
Comment (jasonp123) [inline src/Container/Lifetime/LifetimeContainer.cs]: Shouldn't the exception returned be consistent?

Comment (jasonp123) [reply src/Container/Lifetime/LifetimeContainer.cs]: What I mean is, shouldn't it always be an AggregateException, then the consumer can flatten

**Comment by ENikS** [comment]
Could you replace Moq with POCO implementation? I would prefer not to depend on Moq.

File: src/Container/Lifetime/LifetimeContainer.cs (hunk 1 above)
Comment (ENikS) [reply src/Container/Lifetime/LifetimeContainer.cs]: The way you do it is fine

Comment (hanabi1224) [reply src/Container/Lifetime/LifetimeContainer.cs]: @jasonp123  For best backward compatibility,  when there's single exception, the type of exception being thrown remains the same before and after this change

**Pull request merged**